src/HeaderBag.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Aidantwoods\SecureHeaders;
+
+class HeaderBag
+{
+    protected $headers = array();
+
+    public function __construct(array $headers = array())
+    {
+        // Send all headers through `replace` to make sure they are properly lower-cased
+        foreach ($headers as $name => $value)
+        {
+            $this->replace($name, $value);
+        }
+    }
+
+    public static function fromHeaderLines(array $lines)
+    {
+        $headers = array();
+
+        foreach ($lines as $line)
+        {
+            list($name, $value) = explode(': ', $line, 2);
+
+            $headers[$name] = $value;
+        }
+
+        return new static($headers);
+    }
+
+    public function has($name)
+    {
+        return array_key_exists(strtolower($name), $this->headers);
+    }
+
+    public function add($name, $value = '')
+    {
+        if ($this->has($name))
+        {
+            return;
+        }
+
+        $this->replace($name, $value);
+    }
+
+    public function replace($name, $value = '')
+    {
+        $this->headers[strtolower($name)] = $value;
+    }
```

Comment:
Whoops, I think I messed up the difference between `add` and `replace` here. `add` should still add the header if it already exists, just add another equally-named header...